packages/turf-clusters/index.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * Get Cluster
+ *
+ * @param {FeatureCollection|Feature[]} geojson GeoJSON Features
+ * @param {*} filter Filter used on GeoJSON properties to get Cluster
+ * @returns {FeatureCollection} Single Cluster filtered by GeoJSON Properties
+ * @example
+ * var geojson = turf.featureCollection([
+ *     turf.point([0, 0], {cluster: 0, foo: 'null'}),
+ *     turf.point([2, 4], {cluster: 1, foo: 'bar'}),
+ *     turf.point([3, 6], {cluster: 1}),
+ *     turf.point([5, 1], {0: 'foo'}),
+ *     turf.point([4, 2], {'bar': 'foo'})
+ * ]);
+ *
+ * // Create a cluster by Object
+ * var cluster1 = turf.getCluster(geojson, {cluster: 1});
+ *
+ * // Create a cluster by String
+ * var cluster2 = turf.getCluster(geojson, 'cluster');
+ *
+ * // Create a cluster by an Array of Strings
+ * var cluster3 = turf.getCluster(geojson, ['cluster', 'foo']);
+ */
+function getCluster(geojson, filter) {
+    // Validation
+    if (!geojson) throw new Error('geojson is required');
+    if (filter === undefined || filter === null) throw new Error('filter is required');
+    if (geojson.type === 'FeatureCollection') geojson = geojson.features;
+    if (!Array.isArray(geojson)) throw new Error('invalid geojson');
```

Comment:
Originally I had it as `FeatureCollection|Feature[]` (`FeatureCollection|Array<Feature>`), now that we've dropped the Array of features (since it's not a valid GeoJSON), there isn't much complex validation to do which would require using `@turf/invariant`. Using [`geojsonType`](https://github.com/Turfjs/turf/blob/master/packages/turf-invariant/index.js#L74-L89) is already being handled [here](https://github.com/Turfjs/turf/pull/847/files#diff-b96c64619e5de1f4ad32bb28a082aa44R28):
```js
if (geojson.type !== 'FeatureCollection') throw new Error('geojson must be a FeatureCollection');
```